File: backend/api/schemas/member.py
"""Pydantic schemas for member endpoints."""

from __future__ import annotations

from datetime import date

from pydantic import BaseModel, Field

from api.schemas.pagination import PaginatedResponse


class MemberResponse(BaseModel):
    id: str
    name: str
    email: str | None = None
    phone: str | None = None
    status: str
    enrolled_at: date | None = None
    cancelled_at: date | None = None


class MemberListResponse(PaginatedResponse):
    members: list[MemberResponse]


class SignalsResponse(BaseModel):
    dias_sem_treino: int = 0
    queda_frequencia: int = 0
    inadimplencia: int = 0
    queda_duracao: int = 0
    baixa_frequencia: int = 0
    historico_pagamento: int = 0
    aluno_novo: int = 0


class MemberScoreResponse(BaseModel):
    member: MemberResponse
    score: int
    tier: str
    reasons: list[str] = Field(default_factory=list)
    signals: SignalsResponse = Field(default_factory=SignalsResponse)
    computed_at: date

File: frontend/src/types/member.ts
import type { Tier, MemberStatus } from "./dashboard";

export interface MemberResponse {
  id: string;
  name: string;
  email: string | null;
  phone: string | null;
  status: MemberStatus;
  enrolled_at: string | null;
  cancelled_at: string | null;
}

export interface MemberListResponse {
  members: MemberResponse[];
  total: number;
  page: number;
  page_size: number;
}

export interface SignalsResponse {
  dias_sem_treino: number;
  queda_frequencia: number;
  inadimplencia: number;
  queda_duracao: number;
  baixa_frequencia: number;
  historico_pagamento: number;
  aluno_novo: number;
}

export interface MemberScoreResponse {
  member: MemberResponse;
  score: number;
  tier: Tier;
  reasons: string[];
  signals: SignalsResponse;
  computed_at: string;
}
